src/result_processor.c

```diff
@@ -368,6 +371,7 @@ int sorter_Next(ResultProcessorCtx *ctx, SearchResult *r) {
       h->indexResult = NULL;
       sc->pooledResult = mmh_pop_min(sc->pq);
       SearchResult_FreeInternal(sc->pooledResult);
+      DocTable_IncreaseDmdRefCount(h->scorerPrivateData);
```

Comment:
I'd like a helper function which merges pop and decref. Also, we should be prepared to handle a situation where refcount is 1 (in which case it means the document is deleted). thoughts?